lib/WellKnown/WebfingerHandler.php

```diff
@@ -216,8 +222,11 @@ private function getSubjectFromRequest(IRequest $request): string {
 
 		// work around to extract resource:
 		// on some setup (i.e. tests) the data are not available from IRequest
-		parse_str(parse_url($request->getRequestUri(), PHP_URL_QUERY), $query);
-
+		$requestUri = $request->getRequestUri();
+		if ($requestUri !== null) {
+			parse_str(parse_url($requestUri, PHP_URL_QUERY), $query);
```

Comment:
parse_url will return null if there is no query.
See https://www.php.net/manual/fr/function.parse-url.php
parse_str will not accept null.
```suggestion
			parse_str(parse_url($requestUri, PHP_URL_QUERY) ?? '', $query);
```

Reply:
This seems to help, I can use curl to get `/.well-known/webfinger` now and I get back: `""` with HTTP 400.
This still makes the security check in the Administration Settings unhappy, but at least no error is logged anymore.

Reply:
You're right. Thanks, @come-nc.

@paulvt Setup check is being dealt with nextcloud/server#49440